lapack/tpls/KokkosLapack_gesv_tpl_spec_decl.hpp

```diff
@@ -176,8 +188,8 @@ namespace Impl {
         Kokkos::MemoryTraits<Kokkos::Unmanaged>>                              \
         PViewType;                                                            \
                                                                               \
-    static void gesv(const AViewType& A, const BViewType& B,                  \
-                     const PViewType& IPIV) {                                 \
+    static void gesv(const Kokkos::Cuda& /*space*/, const AViewType& A,       \
```

Comment:
I have not done the stream based approached here, the link you give is good but it seems that the approach using a device ID is the one recommended by MAGMA, this might be a reasonable follow-on.